### DISCUSSION TITLE
Frigate - memory leaks and hangs

### DISCUSSION THREAD

**Discussion by Vendo233** [tteck/Proxmox#3409] [Help Needed] [answered]
I`m having issue with Frigate LXC where Frigat stops working and Memory and Swap are 100%. ( memory leak? ) 
What can I do? 
After reboot the LXC works and then in several hours when memory and swap is maxxed out it hangs

**Comment by Vendo233** [comment]
![image](https://github.com/user-attachments/assets/e2f7f62a-2338-492a-bcd7-72e03bc876bd)

**Reply by Chrischi-** [reply]
Do you use opus audio (go2rtc config)? If so, then deactivate audio completely.

Should be fixed with frigate 14.x

**Comment by Citroene** [comment]
i have the same issue, when SWAP is at 100%, the container is unresponsive and frigate UI keeps loading. after reboot everything works for a few days.

**Comment by supershopping** [comment]
I am having the same memory issue issue. Using Frigate LXC.

**Comment by supershopping** [comment]
My issue is related to the high python memory usage. See below discussion.
https://github.com/blakeblackshear/frigate/discussions/10921#discussioncomment-9073711

**Comment by supershopping** [comment]
My question is how to limit memory usage(like to 3GB) with the LXC helper script installation?

**Comment by iz8mbw** [comment]
same issue here since version 0.14:
![image](https://github.com/user-attachments/assets/21095bfd-c18f-4825-89a5-6cf2b2bfa2b7)


I suppose memory leak.

**Comment by Vendo233** [comment, marked as answer]
updating to Frigate 14 fixed the issue.

**Reply by iz8mbw** [reply]
I have this issue since Frigate 0.14 (all was ok with version 0.13)

**Reply by supershopping** [reply]
I have Frigate 14 too. If a small portion of memory( compare with your whole available memory) is assigned to the frigate LXC, the memory leak issue could affect your system.

**Reply by iz8mbw** [reply]
I don't understand.
Are 4 GB of RAM enough?
If yes it must work.
Generally a memory leak is an issue and not a normal behaviour

**Reply by supershopping** [reply]
It's the percent of your total server memory. How much memory does your server have? You can check your system python usage with the info in this post. 
https://github.com/blakeblackshear/frigate/discussions/10921#discussioncomment-10311626

Someone mentioned a possible work around is to limit the memory cap to 3GB in the docker compose file.
